protocol/test/foundry/PipelineConvert.t.sol

```diff
@@ -0,0 +1,1461 @@
+
+        uint256[] memory tokenAmountsIn = new uint256[](2);
+        tokenAmountsIn[0] = 0;
+        tokenAmountsIn[1] = ethAmount;
+
+        vm.prank(users[1]);
+        uint256 lpAmountOut = IWell(C.BEAN_ETH_WELL).addLiquidity(tokenAmountsIn, 0, users[1], type(uint256).max);
+
+        // get new deltaB
+        int256 beforeDeltaB = seasonGetters.poolDeltaBInsta(C.BEAN_ETH_WELL);
+        
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+        uint256 grownStalkBefore = bs.balanceOfGrownStalk(users[1], C.BEAN);
+
+
+        beanToLPDoConvert(amount, stem, users[1]);
+
+        uint256 totalStalkAfter = bs.balanceOfStalk(users[1]);
+
+        // get balance of deposited bdv for this user
+        uint256 bdvBalance = bs.balanceOfDepositedBdv(users[1], C.BEAN_ETH_WELL) * 1e4; // convert to stalk amount
+
+        assertTrue(totalStalkAfter == bdvBalance + grownStalkBefore); // all grown stalk was lost
+    }
+
+    function testFlashloanManipulationLoseGrownStalkBecauseZeroConvertCapacity(uint256 amount) public {
+        amount = bound(amount, 5000e6, 5000e6); // todo: update for range
+
+        // the main idea is that we start at deltaB of zero, so converts should not be possible
+        // we add eth to the well to push it over peg, then we convert our beans back down to lp
+        // then we pull our initial eth back out and we converted when we shouldn't have been able to (if we do in one tx)
+
+        // setup initial bean deposit
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+
+        // mint user 10 eth
+        uint256 ethAmount = 10e18;
+        MockToken(C.WETH).mint(users[1], ethAmount);
+
+        vm.prank(users[1]);
+        MockToken(C.WETH).approve(C.BEAN_ETH_WELL, ethAmount);
+
+        // add liquidity to well
+        uint256[] memory tokenAmountsIn = new uint256[](2);
+        tokenAmountsIn[0] = 0;
+        tokenAmountsIn[1] = ethAmount;
+
+        vm.prank(users[1]);
+        IWell(C.BEAN_ETH_WELL).addLiquidity(tokenAmountsIn, 0, users[1], type(uint256).max);
+
+        uint256 grownStalkBefore = bs.balanceOfGrownStalk(users[1], C.BEAN);
+
+        // convert beans to lp
+        beanToLPDoConvert(amount, stem, users[1]);
+
+        // it should be that we lost our grown stalk from this convert
+
+        uint256 grownStalkAfter = bs.balanceOfGrownStalk(users[1], C.BEAN_ETH_WELL);
+
+        assertTrue(grownStalkAfter == 0); // all grown stalk was lost
+        assertTrue(grownStalkBefore > 0);
+    }
+
+    // double convert uses up convert power so we should be left at no grown stalk after second convert
+    // (but still have grown stalk after first convert)
+    function testFlashloanManipulationLoseGrownStalkBecauseDoubleConvert(uint256 amount) public {
+        amount = bound(amount, 1000e6, 1000e6); // todo: update for range
+
+        // do initial pump update
+        updateMockPumpUsingWellReserves(C.BEAN_ETH_WELL);
```

Comment:
why is a mock pump update needed here?

Reply:
This one appears unnecessary, removed and test still works fine: https://github.com/BeanstalkFarms/Beanstalk/commit/4e14d49792b50096281af79ce357f6a014636cd3